lib/python/picongpu/pypicongpu/grid.py

```diff
@@ -38,8 +39,27 @@ def get_cfg_str(self) -> str:
         return literal_by_boundarycondition[self]
 
 
-@typeguard.typechecked
-class Grid3D(RenderedObject):
+def serialise_vec(value) -> dict:
+    return dict(zip("xyz", value))
```

Comment:
```suggestion
    return dict(zip("xyz", value, strict=True))
```
Otherwise, iterable longer or shorter than 3 will not fail. Requires python 3.10 or higher.

Reply:
Or is the wrong length of the iterator already caught automatically by pydantic with  `tuple[int,int,int]`?

Reply:
That's likely true. Also, I explicitly wrote it like this (quite a while ago, this is just moved from further below) because it would already apply to the (not-yet-supported) 2D case. Not a strong argument for or against it though, so I'd be willing to change.